### PR TITLE
fix slow scatternd

### PR DESCRIPTION
**Pull request by irkjero**
fix slow scatter nd

### CODE REVIEW

File: onnx2torch/node_converters/scatter_nd.py [hunk 1]
```diff
@@ -19,13 +18,17 @@ def forward(self, data: torch.Tensor, indices: torch.Tensor, updates: torch.Tens
...
-        update_indices = indices.shape[:-1]
-        for idx in np.ndindex(update_indices):
-            output[tuple(indices[idx])] = updates[idx]
 
         if torch.onnx.is_in_onnx_export():
             return _ScatterNDExportToOnnx.set_output_and_apply(output, data, indices, updates)
 
+        ind_dim = indices.dim()
+        # last dimension is a partial-index into data
+        indices = indices.reshape((-1, indices.shape[-1])).T.tolist()
```
Comment (ivkalgin) [inline onnx2torch/node_converters/scatter_nd.py:27]: Transpose can be skipped
```python
indices = indices.reshape((indices.shape[-1], -1)).tolist()
```

File: onnx2torch/node_converters/scatter_nd.py [hunk 2]
```diff
@@ -19,13 +18,17 @@ def forward(self, data: torch.Tensor, indices: torch.Tensor, updates: torch.Tens
...
+        # update.shape = indices.shape[0:ind_dim-1] ++ data.shape[indices.shape[-1]:data.dim()-1]
+        updates = updates.reshape((-1, *list(updates.shape[ind_dim - 1:])))
```
Comment (ivkalgin) [inline onnx2torch/node_converters/scatter_nd.py:29]: list can be skipped
```python
updates = updates.reshape((-1, *updates.shape[ind_dim - 1:]))
```

Comment (irkjero) [reply onnx2torch/node_converters/scatter_nd.py]: Done.

File: onnx2torch/node_converters/scatter_nd.py (hunk 1 above)
Comment (irkjero) [reply onnx2torch/node_converters/scatter_nd.py:27]: indices.reshape((indices.shape[-1], -1)) != indices.reshape((-1, indices.shape[-1])).T

We need not only needed shape, but and swapaxes. Here 'indices.reshape((indices.shape[-1], -1))' we just unroll last dim, but 'indices.reshape((-1, indices.shape[-1])).T' here we have tensor with indices for each dim in data.

**Review by ivkalgin — APPROVED** [review]

**Pull request merged**